eras/conway/impl/testlib/Test/Cardano/Ledger/Conway/Imp/UtxosSpec.hs

```diff
@@ -0,0 +1,77 @@
+{-# LANGUAGE NumericUnderscores #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Test.Cardano.Ledger.Conway.Imp.UtxosSpec where
+
+import Cardano.Ledger.Address (Addr (..))
+import Cardano.Ledger.BaseTypes (Inject (..), Network (..), StrictMaybe (..), maybeToStrictMaybe)
+import Cardano.Ledger.Coin (Coin (..))
+import Cardano.Ledger.Conway.Core (
+  AlonzoEraScript (..),
+  AlonzoEraTxOut (..),
+  BabbageEraTxBody (..),
+  BabbageEraTxOut (..),
+  EraTx (..),
+  EraTxBody (..),
+  EraTxOut (..),
+  txIdTx,
+ )
+import Cardano.Ledger.Credential (Credential (..), StakeReference (..))
+import Cardano.Ledger.Plutus (Data (..), hashData)
+import Cardano.Ledger.TxIn (mkTxInPartial)
+import qualified Data.Sequence.Strict as SSeq
+import qualified Data.Set as Set
+import Lens.Micro ((&), (.~))
+import qualified PlutusLedgerApi.V1 as P1
+import Test.Cardano.Ledger.Conway.ImpTest (ImpTestState, ShelleyEraImp, freshKeyHash, impAddPlutusScript, lookupKeyPair, submitTxAnn, submitTxAnn_)
+import Test.Cardano.Ledger.Core.KeyPair (mkAddr)
+import Test.Cardano.Ledger.Imp.Common
+import Test.Cardano.Ledger.Plutus (PlutusArgs (..), ScriptTestContext (..))
+import Test.Cardano.Ledger.Plutus.Examples (guessTheNumber3)
+
+spec ::
+  forall era.
+  ( ShelleyEraImp era
+  , BabbageEraTxBody era
+  ) =>
+  SpecWith (ImpTestState era)
+spec =
+  describe "UTXOS" $ do
+    it "can use reference scripts" $ do
+      let spendDatum = P1.I 3
+      shSpending <-
+        impAddPlutusScript
+          ScriptTestContext
+            { stcScript = guessTheNumber3
+            , stcArgs =
+                PlutusArgs
+                  { paSpendDatum = Just spendDatum
+                  , paData = spendDatum
+                  }
+            }
+      kpPayment <- lookupKeyPair =<< freshKeyHash
+      kpStaking <- lookupKeyPair =<< freshKeyHash
+      let
+        refTxOut =
+          mkBasicTxOut (mkAddr (kpPayment, kpStaking)) (inject $ Coin 100)
+            & referenceScriptTxOutL .~ maybeToStrictMaybe (fromPlutusScript <$> mkPlutusScript guessTheNumber3)
+        scriptLockedTxOut =
+          mkBasicTxOut
+            (Addr Testnet (ScriptHashObj shSpending) StakeRefNull)
+            (inject $ Coin 1_000_000)
+            & dataHashTxOutL .~ SJust (hashData @era $ Data spendDatum)
+      producingTx <-
+        submitTxAnn "Producing transaction" $
+          mkBasicTx mkBasicTxBody
+            & bodyTxL . outputsTxBodyL
+              .~ SSeq.fromList
+                [ refTxOut
+                , scriptLockedTxOut
+                ]
+      let producingTxId = txIdTx producingTx
+      submitTxAnn_ "Consuming transaction" $
```

Comment:
We need some more checks here, because from what I can see in the Conway.TxInfo implementation reference scripts are still not enabled for PlutusV1 

Can you please add a check that verifies that there was no witness added for that script, effectively ensuring that it was indeed the reference script that was used.
```suggestion
      tx <- submitTxAnn "Consuming transaction" $
```